packages/slice-machine/src/modules/preview/index.ts

```diff
@@ -0,0 +1,284 @@
+      return {
+        ...state,
+        setupStatus: {
+          ...state.setupStatus,
+          iframe: "ko",
+        },
+        isWaitingForIframeCheck: false,
+      };
+    case getType(toggleSetupDrawerStepCreator):
+      return {
+        ...state,
+        setupDrawer: {
+          ...state.setupDrawer,
+          openedStep:
+            state.setupDrawer.openedStep === action.payload.stepNumber
+              ? NoStepSelected
+              : action.payload.stepNumber,
+        },
+      };
+    case getType(checkPreviewSetupCreator.success):
+      return {
+        ...state,
+        setupStatus: {
+          ...state.setupStatus,
+          ...action.payload.setupStatus,
+        },
+      };
+    case getType(closeSetupPreviewDrawerCreator):
+      return {
+        ...state,
+        setupDrawer: {
+          openedStep: NoStepSelected,
+          isOpen: false,
+        },
+      };
+    default:
+      return state;
+  }
+};
+
+// Sagas
+function* checkSetupSaga(
+  action: ReturnType<typeof checkPreviewSetupCreator.request>
+) {
+  try {
+    const { data: setupStatus }: { data: PreviewCheckResponse } = yield call(
+      checkPreviewSetup
+    );
+
+    // All the backend checks are ok ask for the frontend Iframe check
+    if ("ok" === setupStatus.manifest && "ok" === setupStatus.dependencies) {
+      yield put(
+        checkPreviewSetupCreator.success({
+          setupStatus: { iframe: null, ...setupStatus },
+        })
+      );
+      yield put(connectToPreviewIframeCreator.request());
+      const { timeout, iframeCheckKO, iframeCheckOk } = yield race({
+        iframeCheckOk: take(getType(connectToPreviewIframeCreator.success)),
+        iframeCheckKO: take(getType(connectToPreviewIframeCreator.failure)),
+        timeout: delay(2500),
+      });
+
+      if (iframeCheckOk && action.payload.callback) {
+        action.payload.callback();
+        return;
+      }
+
+      if (timeout) {
+        yield put(connectToPreviewIframeCreator.failure());
+        yield call(failCheckSetupSaga);
+        return;
+      }
+
+      if (iframeCheckKO) {
+        yield call(failCheckSetupSaga);
+        return;
+      }
+      return;
+    }
+
+    // All the backend checks are ko and the request is coming from the "start"
+    const isTheFirstTime =
+      "ko" === setupStatus.manifest &&
+      "ko" === setupStatus.dependencies &&
+      action.payload.withFirstVisitCheck;
+
+    if (isTheFirstTime) {
+      yield put(openSetupPreviewDrawerCreator({}));
+      return;
+    }
+
+    yield put(
+      checkPreviewSetupCreator.success({
+        setupStatus: { iframe: null, ...setupStatus },
+      })
+    );
+    yield call(failCheckSetupSaga);
+  } catch (error) {
+    yield put(checkPreviewSetupCreator.failure(error));
+  }
+}
+
+function* failCheckSetupSaga() {
+  const framework: Frameworks = yield select(getFramework);
+
+  if (![Frameworks.next, Frameworks.nuxt].includes(framework)) {
```

Comment:
Same here, we can use `previewIsSupported`

Reply:
Done :)